packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -848,186 +827,71 @@ export const useArbTokenBridge = (
     }
 
     const bridgeTokensToAdd: ContractStorage<ERC20BridgeToken> = {}
-    const { name, symbol, balance, decimals } = await getL1TokenData({
+    const { name, symbol, decimals } = await getL1TokenData({
       account: walletAddress,
       erc20L1Address: l1Address,
       l1Provider: l1.provider,
       l2Provider: l2.provider
     })
 
-    l1TokenBalance = balance
-
-    try {
-      // check if token is deployed at l2 address; if not this will throw
-      const { balance } = await getL2TokenData({
-        account: walletAddress,
-        erc20L2Address: l2Address,
-        l2Provider: l2.provider
-      })
-      l2TokenBalance = balance
-    } catch (error) {
-      console.info(`no L2 token for ${l1Address} (which is fine)`)
-
-      l2Address = undefined
-    }
-
     const isDisabled = await l1TokenIsDisabled(l1Address)
 
     if (isDisabled) {
       throw new TokenDisabledError('Token currently disabled')
     }
 
-    bridgeTokensToAdd[l1Address.toLowerCase()] = {
+    const l1AddressLowerCased = l1Address.toLowerCase()
+    bridgeTokensToAdd[l1AddressLowerCased] = {
       name,
       type: TokenType.ERC20,
       symbol,
-      address: l1Address.toLowerCase(),
-      l2Address,
+      address: l1AddressLowerCased,
+      l2Address: l2Address?.toLowerCase(),
       decimals
     }
 
     setBridgeTokens(oldBridgeTokens => {
       return { ...oldBridgeTokens, ...bridgeTokensToAdd }
     })
-    setErc20Balances(oldBridgeBalances => {
-      const newBal = {
-        [l1Address.toLowerCase()]: {
-          balance: l1TokenBalance,
-          arbChainBalance: l2TokenBalance
-        }
-      }
-      return { ...oldBridgeBalances, ...newBal }
-    })
-    return l1Address
-  }
-
-  const expireCache = (): void => {
-    clearERC20Cache()
-    clearERC721Cache()
-  }
 
-  useEffect(() => {
-    const tokensToAdd = [
-      ...new Set([...defaultTokenList].map(t => t.toLocaleLowerCase()))
-    ].filter(tokenAddress => !tokenBlackList.includes(tokenAddress))
-    if (autoLoadCache) {
-      Promise.all(
-        tokensToAdd.map(address => {
-          return addToken(address).catch(err => {
-            console.warn(`invalid cache entry erc20 ${address}`)
-            console.warn(err)
-          })
-        })
-      ).then(values => {
-        setERC20Cache(values.filter((val): val is string => !!val))
-      })
+    updateErc20L1Balance([l1AddressLowerCased])
+    if (l2Address) {
+      updateErc20L2Balance([l2Address.toLowerCase()])
     }
-  }, [])
+  }
 
   const updateTokenData = useCallback(
```

Comment:
`updateTokenData` should be updating balances for the token, which I don't think it's doing anymore? It's called on deposit/withdraw, when the balance changes.

Reply:
It didn't seem to cause issue when I was testing, but I have added the update to be safe